node/task.ts

```diff
@@ -1775,6 +1803,218 @@ export class CodeCoverageEnabler {
+/**
+ * Instruction for the agent to update the PATH environment variable.
+ * The specified directory is prepended to the PATH.
+ * The updated environment variable will be reflected in subsequent tasks.
+ * 
+ * @param path      Local directory path.
+ * @returns         void
+ */
+export function prependPath(path: string) {
+    assertAgent("2.115.0");
+    command("task.prependpath", null, path);
+}
+
+/**
+ * Upload and attach summary markdown to current timeline record.
+ * This summary shall be added to the build/release summary and 
+ * not available for download with logs.
+ * 
+ * @param path      Local directory path.
+ * @returns         void
+ */
+export function uploadSummary(path: string) {
+    command("task.uploadsummary", null, path);
+}
+
+/**
+ * Upload and attach attachment to current timeline record.
+ * These files are not available for download with logs.
+ * These can only be referred to by extensions using the type or name values. 
+ * 
+ * @param type      Attachment type.
+ * @param name      Attachment name.
+ * @param path      Attachment path.
+ * @returns         void
+ */
+export function addAttachment(type: string, name: string, path: string) {
+    command("task.addattachment", { "type": type, "name": name }, path);
+}
+
+/**
+ * Set an endpoint field with given value.
+ * Value updated will be retained in the endpoint for 
+ * the subsequent tasks that execute within the same job.
+ * 
+ * @param id      Endpoint id.
+ * @param field   FieldType enum of AuthParameter, DataParameter or Url.
+ * @param key     Key.
+ * @param value   Value for key or url.
+ * @returns       void
+ */
+export function setEndpoint(id: string, field: FieldType, key: string, value: string) {
+    command("task.setendpoint", { "id": id, "field": field, "key": key }, value);
+}
+
+/**
+ * Set progress and current operation for current task.
+ * 
+ * @param percent           Percentage of completion.
+ * @param currentOperation  Current pperation. 
+ * @returns                 void
+ */
+export function setProgress(percent: number, currentOperation: string) {
+    command("task.setprogress", { "value": `${percent}` }, currentOperation);
+}
+
+/**
+ * Indicates whether to write the logging command directly to the host or to the output pipeline.
+ *
+ * @param id            Timeline record Guid.
+ * @param parentId      Parent timeline record Guid.
+ * @param recordType    Record type.
+ * @param recordName    Record name.
+ * @param order         Order of timeline record.
+ * @param startTime     Start time.
+ * @param finishTime    End time.
+ * @param progress      Percentage of completion.
+ * @param state         TaskState enum of Unknown, Initialized, InProgress or Completed.
+ * @param result        TaskResult enum of Succeeded, SucceededWithIssues, Failed, Cancelled or Skipped.
+ * @param message       current operation
+ * @returns             void
+ */
+export function logDetail(id: string, parentId: string, recordType: string,
```

Comment:
Most of these properties should be optional - see here https://github.com/Microsoft/azure-pipelines-tasks/blob/master/docs/authoring/commands.md